### PR TITLE
Small optimization for discovery label

### PR DESCRIPTION
**Pull request by lolodomo**

### CODE REVIEW

**Comment by lolodomo** [comment]
Wait a minute...

**Comment by lolodomo** [comment]
Ok, there was a bug in primaryDeviceTypeForEndpoint with a wrong operator in the if condition. It is fixed now.

**Comment by lolodomo** [comment]
Here is the new result in inbox with the Switchbot Hub 2 matter device:

image

It is perfect for endpoints 2 and 3.
For endpoints 7,8,9, temperature is chosen but it contains in fact temperature and humidity. The first of both is chosen, OK.

It also leads to that for the thing:

image

**Comment by lolodomo** [comment]
Another (better) option would be to combine them when there are several like for example "Temperature Sensor + Humidity Sensor". WDYT ?
It could be done later, it is already not so bad now.

File: bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/util/MatterLabelUtils.java [hunk 1]
```diff
@@ -58,7 +62,7 @@ public static String labelForNode(Endpoint root) {
         if (!nodeLabel.isEmpty()) {
             label = nodeLabel;
         } else {
-            label = vendorName + " " + productName;
+            label = productName.startsWith(vendorName) ? productName : vendorName + " " + productName;
```
Comment (digitaldan) [inline bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/util/MatterLabelUtils.java:65]: 👍 I noticed yesterday  that Apple Home does something like  this as well, looking for duplicate labels, so if i add "Office Lights" to a room called "Office", it changes the name of the device to just "Lights" when in that group.

**Comment by digitaldan** [comment]
> It could be done later, it is already not so bad now.

I think that makes sense,  my only worry was endpoint label getting too long depending on how many device types are listed.

**Review by digitaldan — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
> I think that makes sense, my only worry was endpoint label getting too long depending on how many device types are listed.

We could limit to max 2 or 3 for example to avoid very long label.

**Pull request merged**